internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -58,10 +59,9 @@ trait WithPattern { protected def Pattern: Regex }
 import java.lang.{ Long => BoxedLong }
 
 /** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash(val value: Array[Byte]) extends StampBase {
-  val hexHash = IOHash.toHex(value)
+final class Hash(val hexHash: String) extends StampBase {
```

Comment:
Yes, I'm sure. Scalac's backend transforms `==` into a ref equality check + value equality check. That's why your snippet shows this behaviour. So if two strings point to the same memory pointer (or they are interned) will be short-circuit `==`'s implementation.